packages/framework/presence/src/presenceStates.ts

```diff
@@ -303,7 +303,14 @@ class PresenceStatesImpl<TSchema extends PresenceStatesSchema>
 		content: TSchemaAdditional,
 	): PresenceStates<TSchema & TSchemaAdditional> {
 		for (const [key, nodeFactory] of Object.entries(content)) {
-			this.add(key, nodeFactory);
+			if (key in this.nodes) {
+				const node = unbrandIVM(this.nodes[key]);
+				if (!(node instanceof nodeFactory.instanceBase)) {
+					throw new TypeError(`State "${key}" previously created by different value manager.`);
```

Comment:
I think I understand what you are doing here, but how does this scenario play out?

Reply:
Hitting a problem is a fairly advanced case.
Kash demonstrated that it is easy to exercise this whole pattern with React without being careful. But wouldn't be likely wrong and fail this. How to fail this:
1. Have two code components that want to share the same states workspace.
2. One uses schema:
   `const schema = { pointers: Latest({ mouse: {x: 0, y: 0}, pen: {x: 0, y: 0}}); }`
3. Other uses schema:
   `const schema = { pointers: LatestMap({ mouse: {x: 0, y: 0}, pen: {x: 0, y: 0}}); }`
4. Both call `presence.getStates("name:our-stuff", schema)`
One of them will activate first, but the value managers will be different. We need to fail whomever goes second.

It might be more likely that two components try to use the same lazily crafted workspace address ("name:our-stuff") above and have very different. but conflicting schemas. We want to fail on that.